src/FileSystem/FileSystemInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Akeneo\Pim\ApiClient\FileSystem;
+
+use Akeneo\Pim\ApiClient\Exception\InvalidArgumentException;
+
+/**
+ * Manipulates files for the API.
+ *
+ * @author    Laurent Petard <laurent.petard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface FileSystemInterface
+{
+    /**
+     * Gets the resource of a file from its path.
+     *
+     * @param string $filePath Path of the file
+     *
+     * @throws InvalidArgumentException if the file doesn't exists or is not readable
```

Comment:
create a dedicated exception seems a good idea as you have a dedicated interface (and exception are not BC break in php)